### PR TITLE
Add support to anchor fm embeds

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4152

Related to https://github.com/opencollective/opencollective-api/pull/5820

Should only be merged after, https://github.com/opencollective/opencollective-frontend/pull/6042

## Screencast

Peek 2021-04-03 16-21

### CODE REVIEW

File: components/HTMLContent.js [hunk 1]
```diff
@@ -120,6 +120,11 @@ const HTMLContent = styled(({ content, collapsable, sanitize, ...props }) => {
         height: 100%;
       }
     }
+    &[data-trix-content-type='--embed-iframe-anchorFm'] {
+      iframe {
+        min-height: 300px;
```
Comment (Betree) [inline components/HTMLContent.js:125]: AnchorFm has a min-height set on its content. This prevents adding a scrollbar on the iframe.

File: components/RichTextEditor.js [hunk 2]
```diff
@@ -330,58 +330,82 @@ export default class RichTextEditor extends React.Component {
...
     }
   };
 
-  parseVideoLink = videoLink => {
+  parseServiceLink = videoLink => {
     const regexps = {
       youtube: new RegExp(
-        '(?:https?://)?(?:www.)?youtu(?:.be/|be.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
-        'ig',
+        '(?:https?://)?(?:www\\.)?youtu(?:\\.be/|be\\.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
```
Comment (Betree) [inline components/RichTextEditor.js:365]: Edited YouTube and Vimeo regexps to escape dots. before that, `https://vimeoxcom/xxxx` was considered as a valid Vimeo link.

File: components/RichTextEditor.js [hunk 3]
```diff
@@ -330,58 +330,82 @@ export default class RichTextEditor extends React.Component {
...
+        'i',
       ),
-      // vimeo: new RegExp(
-      //   '(http|https)?://(www.)?vimeo.com/(?:channels/(?:\\w+/)?|groups/([^/]*)/videos/|)(\\d+)(?:|/?)',
-      // ),
+      vimeo: new RegExp(
+        '(http|https)?://(www\\.)?vimeo\\.com/(?:channels/(?:\\w+/)?|groups/([^/]*)/videos/|)(\\d+)(?:|/?)',
+      ),
+      anchorFm: /^(http|https)?:\/\/(www\.)?anchor\.fm\/([^/]+)\/(embed)?(\/episodes\/)?([^/]+)?/,
```
Comment (Betree) [inline components/RichTextEditor.js:371]: Updated to better validate links. See https://regex101.com/r/ScxxtS/1

**Comment by SudharakaP** [comment]
@Betree : Thanks for doing the enhancements. 😺

**Pull request merged**